prusti-server/src/process_verification.rs

```diff
@@ -4,24 +4,110 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-use crate::{VerificationRequest, ViperBackendConfig};
+use crate::{VerificationRequest, ViperBackendConfig, ServerMessage};
 use log::info;
 use prusti_common::{
     config,
     report::log::{report, to_legal_file_name},
     vir::{program_normalization::NormalizationInfo, ToViper},
     Stopwatch,
 };
-use std::{fs::create_dir_all, path::PathBuf};
+use std::{fs::create_dir_all, path::PathBuf, thread, sync::{mpsc, Arc, self}};
 use viper::{
-    smt_manager::SmtManager, Cache, VerificationBackend, VerificationContext, VerificationResult,
+    smt_manager::SmtManager, PersistentCache, Cache, VerificationBackend, VerificationResult, VerificationResultType, Viper, VerificationContext, jni_utils::JniUtils
 };
+use viper_sys::wrappers::viper::*;
+use viper_sys::wrappers::java;
+use std::time;
+use futures::{stream::Stream, lock};
 
-pub fn process_verification_request<'v, 't: 'v>(
-    verification_context: &'v VerificationContext<'t>,
-    mut request: VerificationRequest,
+enum ServerRequest {
+    Verification(VerificationRequest),
+    SaveCache,
+}
+
+pub struct VerificationRequestProcessing {
+    mtx_rx_servermsg: lock::Mutex<mpsc::Receiver<ServerMessage>>,
+    mtx_tx_verreq: sync::Mutex<mpsc::Sender<ServerRequest>>,
+}
+
+// one structure that lives for all the requests and has a single thread working on all the
+// requests sequentially
+// on reception of a verification request, we send it through a channel to the already running
+// thread
```

Comment:
Use `///` for [doc comments](https://doc.rust-lang.org/rust-by-example/meta/doc.html)